agent/agent_test.go

```diff
@@ -186,7 +186,7 @@ func TestAgent_Stats_Magic(t *testing.T) {
 				// If it isn't, it's set to -1.
 				s.ConnectionMedianLatencyMS >= 0
 		}, testutil.WaitLong, testutil.IntervalFast,
-			"never saw stats: %+v", s,
+			"never saw stats",
```

Comment:
This gives misleading information since the pointer is passed on the initial call to `Eventually`.  It is not updated since `<-stats` returns a new pointer, rather than updating the data at the old pointer.  Changed in favor of logging the relevant info as we get it.